### PR TITLE
[AArch64][CodeGen] Fix illegal register aliasing bug for mops instrs

### PR DESCRIPTION
**Pull request by nasherm**
A bug was found where mops instructions were being generated that aliased the source and size registers. This is unpredictable behaviour. This patch usess the earlyclobber constraint on the input source register so that it doesn't alias with the size register. Also a test is introduced which tests affected instructions can't violate this constraint.

### CODE REVIEW

**Review by Stylie777 — COMMENTED** [review]
Are there any LLVM IR tests that can be added for this? As this is a CodeGen issue we should add these.

**Comment by nasherm** [comment]
> I think there should also be some LLVM-Lit tests that check the IR CodeGen. The llvm-mc test checks any assembly which the user inputs into the compiler, but if an ACLE Intrinsic is used this can be checked differently within the tests.
> 
> The tests can be written in a way that the output is check not to be a certain value/register. See #77770 for an example of how a test can ensure early-clobber was being respected within IR and CodeGen.

I've written an IR test and updated the PR

File: llvm/test/CodeGen/AArch64/mops-register-alias.ll [hunk 1]
```diff
@@ -1,26 +1,14 @@
...
-; CHECK-NEXT:    .cfi_def_cfa_offset 80
-; CHECK-NEXT:    movi v0.16b, #64
-; CHECK-NEXT:    mov x8, sp
-; CHECK-NEXT:    mov w9, #64 // =0x40
-; CHECK-NEXT:    mov w10, #64 // =0x40
-; CHECK-NEXT:    add x8, x8, #64
-; CHECK-NEXT:    stp q0, q0, [sp]
-; CHECK-NEXT:    stp q0, q0, [sp, #32]
-; CHECK-NEXT:    setp [x{{[0-9]+}}]!, x{{[0-9]+}}!, x{{[0-9]+}}
+; CHECK:         setp [x{{[0-9]+}}]!, x{{[0-9]+}}!, x{{[0-9]+}}
```
Comment (Stylie777) [inline llvm/test/CodeGen/AArch64/mops-register-alias.ll:6]: Do we care about the lines that have been removed here? I would like to understand the justification of removing the check of the setup for the MOPS instruction.

**Review by Stylie777 — COMMENTED** [review]
One more comment from me.

File: llvm/test/CodeGen/AArch64/mops-register-alias.ll (hunk 1 above)
Comment (nasherm) [reply llvm/test/CodeGen/AArch64/mops-register-alias.ll:6]: I wanted to make what we're testing the focus

**Review by Stylie777 — APPROVED** [review]

**Comment by Stylie777** [comment]
Approved - But worth getting a 2nd set of eyes over it.

**Review by ostannard — APPROVED** [review]
LGTM

**Pull request merged**